### PR TITLE
Add quick start guides for new adopters

### PR DESCRIPTION
**Pull request by michael-valdron**
## What does this PR do / why we need it

Create quick start guides for writing devfiles using odo v3 and Eclipse Che. This is to give the new adopter a practical exercise for learning how to write a simple devfile.

**Additional changes**

- A recent update to the `hotReloadCapable` attribute description in the spec has updated the comments in the spec models
 
## Which issue(s) does this PR fix

Fixes #?

fixes devfile/api#1057

## PR acceptance criteria

- [ ] Unit Tests
- [ ] E2E Tests
- [X] Documentation
_Update the [sidebar](https://github.com/devfile/devfile-web/tree/main/apps/landing-page#configuring-navigation) if there is a new file added or an existing filename is changed_

### CODE REVIEW

File: libs/docs/src/navigation/2.2.0-alpha.yaml
Comment (kim-tsao) [inline libs/docs/src/navigation/2.2.0-alpha.yaml:24]: I think we can remove `2.2.0-alpha` since we are on `2.2.1-alpha` now

**Comment by maysunfaisal** [comment]
Might be a good idea to provide link to Sandbox and also instructions to Che setup

File: libs/docs/src/docs/2.1.0/helloworld-example-che.md
Comment (mike-hoang) [inline libs/docs/src/docs/2.1.0/helloworld-example-che.md:17]:
```suggestion
1. For the hello world example, we will create a simple [hello world Express.js](https://expressjs.com/en/starter/hello-world.html) application
```

File: libs/docs/src/docs/2.1.0/helloworld-example-che.md
Comment (mike-hoang) [inline libs/docs/src/docs/2.1.0/helloworld-example-che.md:144]: Would it be possible to turn this into a diff fence? It would help highlight what to edit in the devfile

File: libs/docs/src/docs/2.1.0/helloworld-example-che.md
Comment (mike-hoang) [inline libs/docs/src/docs/2.1.0/helloworld-example-che.md:358]:
```suggestion
14. Under the 'EXPLORER', expand the 'ENDPOINTS' panel and copy the `http-3000` endpoint URL
```

File: libs/docs/src/docs/2.1.0/helloworld-example-che.md
Comment (mike-hoang) [inline libs/docs/src/docs/2.1.0/helloworld-example-che.md:8]:
```suggestion
## Prerequisites
```

File: libs/docs/src/docs/no-version/helloworld-example-odo.md
Comment (mike-hoang) [inline libs/docs/src/docs/no-version/helloworld-example-odo.md:8]:
```suggestion
## Prerequisites
```

File: libs/docs/src/docs/no-version/helloworld-example-odo.md
Comment (mike-hoang) [inline libs/docs/src/docs/no-version/helloworld-example-odo.md:13]: We do call out that a cluster is required, but I accidentally glossed over this point when I went through the guide. Maybe we can add an optional step to start a minikube cluster?

File: libs/docs/src/navigation/2.1.0.yaml
Comment (mike-hoang) [inline libs/docs/src/navigation/2.1.0.yaml:17]: Should we name it something like "Quick start with odo v3" instead? 

Other options are to put the che and odo guides within the Developing with devfiles page or create a new Quick start page for them

Comment (michael-valdron) [reply libs/docs/src/navigation/2.1.0.yaml:17]: > Should we name it something like "Quick start with odo v3" instead?
> 
> Other options are to put the che and odo guides within the Developing with devfiles page or create a new Quick start page for them

Yeah I like this title better let's use it, I will also make this change for Eclipse Che too, "Quick start with Eclipse Che"

File: libs/docs/src/docs/2.1.0/helloworld-example-che.md
Comment (michael-valdron) [reply libs/docs/src/docs/2.1.0/helloworld-example-che.md]: > Would it be possible to turn this into a diff fence? It would help highlight what to edit in the devfile

@mike-hoang Changed some of these to diff fences and added styles for highlighting, let me know if you have any feedback on these changes.

File: libs/docs/src/docs/no-version/helloworld-example-odo.md
Comment (michael-valdron) [reply libs/docs/src/docs/no-version/helloworld-example-odo.md]: > > We do call out that a cluster is required, but I accidentally glossed over this point when I went through the guide. Maybe we can add an optional step to start a minikube cluster?
> 
> I can add this step for the `odo` guide.

Actually I can add this in both.

**Comment by michael-valdron** [comment]
Waiting on reviews from members of odo and Eclipse Che teams.

**Review by mike-hoang — APPROVED** [review]

File: libs/docs/src/docs/no-version/quickstart-odo.md
Comment (kadel) [inline libs/docs/src/docs/no-version/quickstart-odo.md:11]: Why is `kubectl` or `oc` as a prerequisite? Odo doesn't require either of those and this guide doesn't seem to use them as well.

File: libs/docs/src/docs/no-version/quickstart-odo.md
Comment (kadel) [inline libs/docs/src/docs/no-version/quickstart-odo.md:12]: This guide uses only `odo dev` and devfile being created as part of this doesn't have `image` component so Docker is not required here.

File: libs/docs/src/docs/no-version/quickstart-odo.md
Comment (kadel) [inline libs/docs/src/docs/no-version/quickstart-odo.md:56]: if we are showing snippets of the devfile content, it might be better to make it easier for users and include the actually version

File: libs/docs/src/docs/no-version/quickstart-odo.md
Comment (kadel) [inline libs/docs/src/docs/no-version/quickstart-odo.md:173]: odo doesn't need `kubectl` or `oc`

File: libs/docs/src/docs/no-version/quickstart-odo.md
Comment (kadel) [inline libs/docs/src/docs/no-version/quickstart-odo.md:210]: This should be more visible. We expect most users won't ever start writing the devfile from scratch.

Maybe make it bullet no. 4? And expanding it with how `odo init` can be used.

File: libs/docs/src/docs/no-version/quickstart-odo.md
Comment (michael-valdron) [reply libs/docs/src/docs/no-version/quickstart-odo.md]: > Why is `kubectl` or `oc` as a prerequisite? Odo doesn't require either of those and this guide doesn't seem to use them as well.

Good catch, I missed that `odo` has its own `login` and `logout` commands despite sharing the same login session that `kubectl` and `oc` uses.

File: libs/docs/src/docs/no-version/quickstart-odo.md
Comment (michael-valdron) [reply libs/docs/src/docs/no-version/quickstart-odo.md:63]: > if we are showing snippets of the devfile content, it might be better to make it easier for users and include the actually version

It will include the actual schema version, `<version>` is a placeholder for printing the selected schema version dynamically in that spot. So if you load the site under the `2.2.0` doc pages, `<version>` will be replaced by `2.2.0`.

File: libs/docs/src/docs/2.1.0/quickstart-che.md
Comment (amisevsk) [inline libs/docs/src/docs/2.1.0/quickstart-che.md:29]: This step will not work in Eclipse Che -- https://github.com/eclipse/che/issues/22115. I believe the plan going forward is to allow editing the devfile from within the workspace itself (using the Code editor, for example) but I'm not 100% sure of the status of this feature.

Also, to add, Che currently uses devfile 2.2.0 schema -- the display in the edit page is a bug.

**Review by amisevsk — COMMENTED** [review]
In general, the Eclipse Che portion is somewhat out of date with current/upcoming releases. I'll share this PR with the team for more detailed review.

File: libs/docs/src/docs/2.1.0/quickstart-che.md
Comment (michael-valdron) [reply libs/docs/src/docs/2.1.0/quickstart-che.md:29]: > This step will not work in Eclipse Che -- [eclipse/che#22115](https://github.com/eclipse/che/issues/22115). I believe the plan going forward is to allow editing the devfile from within the workspace itself (using the Code editor, for example) but I'm not 100% sure of the status of this feature.
> 
> Also, to add, Che currently uses devfile 2.2.0 schema -- the display in the edit page is a bug.

Thanks for your review @amisevsk!

Seems like the version of Eclipse Che under Red Hat Dev Spaces is an older version that is still on devfile spec `2.1.0`, I will need to write up revised draft for the current `2.1.0` guide as well as a new guide under `2.2.0` to cover the up to date content: https://github.com/devfile/api/issues/1057#issuecomment-1539069527

**Comment by michael-valdron** [comment]
removed fixes for devfile/api#1113 and devfile/api#1114, these fixes will be in their own PRs

File: libs/docs/src/navigation/2.2.0-alpha.yaml
Comment (michael-valdron) [reply libs/docs/src/navigation/2.2.0-alpha.yaml:24]: This is now being taken care of in PR https://github.com/devfile/devfile-web/pull/89.

File: libs/docs/src/docs/2.1.0/helloworld-example-che.md
Comment (michael-valdron) [reply libs/docs/src/docs/2.1.0/helloworld-example-che.md]: Styling for this is now being taken care of in PR #88.

File: libs/docs/src/docs/no-version/quickstart-odo.md
Comment (kadel) [reply libs/docs/src/docs/no-version/quickstart-odo.md:63]: oh, did not know that. That is really smart 👍

File: libs/docs/src/docs/2.1.0/quickstart-che.md
Comment (michael-valdron) [reply libs/docs/src/docs/2.1.0/quickstart-che.md]: **Update**: https://github.com/devfile/api/issues/1057#issuecomment-1549794771

@amisevsk I need a review of the recent changes when you are able to, thank you.

File: libs/docs/src/docs/2.1.0/quickstart-sandbox.md
Comment (l0rd) [inline libs/docs/src/docs/2.1.0/quickstart-sandbox.md:1]: Unfortunately this procedure is based on the [Dashboard Devfile editor that has been disabled in Eclipse Che 7.65 and that will be disabled starting from next version of OpenShift Dev Spaces](https://github.com/eclipse/che/issues/22115) (the rational behind this can be found [in this epic](https://github.com/eclipse/che/issues/21606)). I would remove this 2.1.0 section to avoid confusions.

**Review by l0rd — CHANGES_REQUESTED** [review]

**Review by l0rd — COMMENTED** [review]
@michael-valdron the Devfile with Eclipse Che getting started article is really nice 👏

**Review by l0rd — APPROVED** [review]

**Review by l0rd — COMMENTED** [review]
👏 LGTM

**Pull request merged**
